src/renderers/shaders/ShaderChunk/common.glsl

```diff
@@ -7,6 +8,26 @@
 #define saturate(a) clamp( a, 0.0, 1.0 )
 #define whiteCompliment(a) ( 1.0 - saturate( a ) )
 
+float average( const in vec3 color ) { return dot( color, vec3( 0.3333 ) ); }
+
+
+struct IncidentLight {
+ 	vec3 color;
+ 	vec3 direction;
+};
+
+struct ReflectedLight {
+ 	vec3 specular;
+ 	vec3 diffuse;
+};
+
```

Comment:
After looking at the entire PR, I am thinking this structure may be cleaner:

    struct ReflectedLight {
        vec3 indirectDiffuse;
        vec3 indirectSpecular;
        vec3 directDiffuse;
        vec3 directSpecular;
    };

    ReflectedLight reflectedLight

It can be changed later if you agree.